File: src/BaseHmiTypes/Scripts/Commands/HmiScriptCallCommand.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Scripts.Commands;

public sealed class HmiScriptCallCommand : HmiScriptCommand
{
    public const string Name = "CallScript";

    public override string CommandName => Name;

    public string? ScriptName { get; set; }

    public HmiScriptLanguage Language { get; set; } = HmiScriptLanguage.Unknown;

    public HmiScriptCallCommand()
    {
        AddArgument(nameof(ScriptName), "Script name", HmiScriptCommandParameterDirection.In);
    }
}

File: typescript/base-hmi-types/src/scripts/commands/HmiScriptCallCommand.ts
import { HmiScriptLanguage } from "../../screens/base/HmiProperty.js";
import { HmiScriptCommand, HmiScriptCommandParameterDirection } from "../HmiScriptCommand.js";

export class HmiScriptCallCommand extends HmiScriptCommand {
  scriptName?: string;
  language = HmiScriptLanguage.Unknown;

  constructor() {
    super();
    this.addArgument("ScriptName", "Script name", HmiScriptCommandParameterDirection.In);
  }

  get commandName(): string {
    return "CallScript";
  }
}
